beacon_node/network/src/beacon_processor/worker/rpc_methods.rs

```diff
@@ -204,6 +204,63 @@ impl<T: BeaconChainTypes> Worker<T> {
         )
     }
 
+    /// Handle a `BlocksByRoot` request from the peer.
+    pub fn handle_light_client_bootstrap(
+        self,
+        executor: TaskExecutor,
+        peer_id: PeerId,
+        request_id: PeerRequestId,
+        request: LightClientBootstrapRequest,
+    ) {
+        // Fetching blocks is async because it may have to hit the execution layer for payloads.
+        executor.spawn(
```

Comment:
This task shouldn't be async, because it doesn't need to fetch any data asynchronously. Constructing the `LightClientBootstrap` is going to be quite CPU and I/O intensive in case of a cache miss, and we want that to happen in the blocking worker thread rather than on the Tokio executor where it will jam things up. i.e. we can just write all this logic in this function directly without the `.spawn`.